### PR TITLE
feat: add missing message types

### PR DESCRIPTION
**Pull request by JustaSqu1d**
## Information

- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Comment by JustaSqu1d** [comment]
Should I add a change-log entry?

**Comment by Lulalaby** [comment]
You can add https://github.com/discord/discord-api-docs/pull/5815 and add a changelog for all :p

File: docs/api/enums.rst [hunk 1]
```diff
@@ -242,6 +242,58 @@ of :class:`enum.Enum`.
         The system message denoting that an context menu command was executed.
 
         .. versionadded:: 2.0
+    .. attribute:: auto_moderation_action
+
+        The system message denoting an action by automod.
+
+        .. versionadded:: 2.3
```
Comment (BobDotCom) [inline docs/api/enums.rst:249]: This should be 2.4?

Comment (Lulalaby) [inline docs/api/enums.rst:249]:
```suggestion
        .. versionadded:: 2.4
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,11 @@ These changes are available on the `master` branch, but have not yet been releas
   `mention_total_limit`; and added the `mention_spam` trigger type.
   ([#1809](https://github.com/Pycord-Development/pycord/pull/1809))
 
+- Added new message types, `interaction_premium_upsell`, `stage_start`, `stage_end`,
```
Comment (BobDotCom) [inline CHANGELOG.md:19]:
```suggestion
  ([#1809](https://github.com/Pycord-Development/pycord/pull/1809))
- Added new message types, `interaction_premium_upsell`, `stage_start`, `stage_end`,
```

File: docs/api/enums.rst (hunk 1 above)
Comment (JustaSqu1d) [reply docs/api/enums.rst]: Nope, it was added earlier in Pycord-Development/pycord#1316 but not documented

File: docs/api/enums.rst [hunk 3]
```diff
@@ -242,6 +242,58 @@ of :class:`enum.Enum`.
         The system message denoting that an context menu command was executed.
 
         .. versionadded:: 2.0
+    .. attribute:: auto_moderation_action
+
+        The system message denoting an action by automod.
+
+        .. versionadded:: 2.4
```
Comment (BobDotCom) [inline docs/api/enums.rst:249]: NVM, this should be 2.3 as it was present prior to this PR.
```suggestion
        .. versionadded:: 2.3
```

Comment (Lulalaby) [reply docs/api/enums.rst:249]: damn

**Review by Lulalaby — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Review by BobDotCom — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Review by ChickenDevs — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**